apps/arweave/src/ar_data_sync.erl

```diff
@@ -3422,6 +3453,37 @@ get_merkle_rebase_threshold() ->
 			Threshold
 	end.
 
+%% @doc Update the replica 2.9 entropy-aligned record of the synced chunks.
+%% It differs from the normal record (ar_data_sync) in that it only registers
+%% the bucket numbers of the synced chunks and records chunks with the same footprint
+%% next to each other. For example, a record may contain intervals 0-10, 1000-1024,
+%% 1020-2048. This means the node has the first 10 chunks of the first entropy footprint,
+%% the last 24 chunks of the first entropy footprint and chunks 20-48 of the second
+%% entropy footprint. These chunks are from the first partition. The offset of the chunks
+%% from the second partition is shifted by the number of chunks in the replica 2.9
+%% entropy generated per partition (which is slightly bigger than the number of chunks
+%% that can fit in the 3.6 TB partition).
+%%
+%% Note that Packing does not have to be replica_2_9. We maintain this record
+%% for any packing so that it is convenient to serve the data to any client.
+add_footprint(PaddedOffset, Packing, StoreID) ->
+	Offset = get_footprint_offset(PaddedOffset),
+	ar_sync_record:add(Offset, Offset - 1, Packing, ar_data_sync_footprints, StoreID).
+
+get_footprint_offset(PaddedOffset) ->
+	FootprintsPerPartition = ?REPLICA_2_9_ENTROPY_COUNT div ?COMPOSITE_PACKING_SUB_CHUNK_COUNT,
+	FootprintSize = ?REPLICA_2_9_ENTROPY_SIZE div ?COMPOSITE_PACKING_SUB_CHUNK_SIZE,
+	Partition = ar_replica_2_9:get_entropy_partition(PaddedOffset),
+	PartitionStartOffset = Partition * FootprintsPerPartition * FootprintSize,
```

Comment:
I think this calculation will cause the partition start offset to drift further and further from the recall partition start offset as we get to higher partitions? If I've read it correctly?

My understanding is that we now have the entropy start and partition start mostly aligned for all partitions (entropy start might start a partial chunk later so we don't split buckets, but no more than one chunk later i think)